src/daft-scan/src/glob.rs

```diff
@@ -104,50 +134,48 @@ impl ScanOperator for GlobScanOperator {
         &[]
     }
 
-    fn num_partitions(&self) -> common_error::DaftResult<usize> {
-        unimplemented!("Cannot get number of partitions -- this will not be implemented.");
-    }
-
-    fn select(self: Box<Self>, columns: &[&str]) -> common_error::DaftResult<ScanOperatorRef> {
-        for c in columns {
-            if self.schema.get_field(c).is_err() {
-                return Err(common_error::DaftError::FieldNotFound(format!(
-                    "{c} not found in {:?}",
-                    self.columns_to_select
-                )));
-            }
-        }
-        let mut to_rtn = self;
-        to_rtn.columns_to_select = Some(columns.iter().map(|s| s.to_string()).collect());
-        Ok(to_rtn)
+    fn can_absorb_filter(&self) -> bool {
+        false
     }
-
-    fn limit(self: Box<Self>, num: usize) -> DaftResult<ScanOperatorRef> {
-        let mut to_rtn = self;
-        to_rtn.limit = Some(num);
-        Ok(to_rtn)
+    fn can_absorb_select(&self) -> bool {
+        false
     }
-
-    fn filter(self: Box<Self>, _predicate: &daft_dsl::Expr) -> DaftResult<(bool, ScanOperatorRef)> {
-        Ok((false, self))
+    fn can_absorb_limit(&self) -> bool {
+        false
     }
 
     fn to_scan_tasks(
-        self: Box<Self>,
+        &self,
+        pushdowns: Pushdowns,
     ) -> DaftResult<Box<dyn Iterator<Item = DaftResult<crate::ScanTask>>>> {
-        let files = run_glob(self.glob_path.as_str(), self.io_config.clone(), None)?;
+        let io_config = match self.storage_config.as_ref() {
+            StorageConfig::Native(cfg) => Arc::new(cfg.io_config.clone().unwrap_or_default()),
+            #[cfg(feature = "python")]
+            StorageConfig::Python(cfg) => Arc::new(cfg.io_config.clone().unwrap_or_default()),
+        };
+        let columns = pushdowns.columns;
+        let limit = pushdowns.limit;
+
+        // Clone to move into closure for delayed execution
+        let storage_config = self.storage_config.clone();
+        let schema = self.schema.clone();
+        let file_format_config = self.file_format_config.clone();
+
+        let files = run_glob(self.glob_path.as_str(), io_config, None)?;
```

Comment:
Does this glob all the files before returning? If so, lets make a note / ticket to address this in the future

Reply:
See: https://github.com/Eventual-Inc/Daft/pull/1577